src/components/screens/dashboard/dashboard-screen.tsx

```diff
@@ -1,18 +1,61 @@
-import { useAtom } from "jotai";
-import { authAtom } from "../../../atoms/auth";
+import { useAtomValue, useSetAtom } from "jotai";
+import { authAtom, userProfileAtom } from "../../../atoms/auth";
+import { useEffect, useState } from "react";
+import { useApi } from "../../../hooks/use-api";
+import { PersonTotalTime } from "../../../generated/client";
+import { errorAtom } from "../../../atoms/error";
 
 /**
  * Dashboard screen component
  * 
  */
-const DashboardScreen = () => {
-  const [auth] = useAtom(authAtom);
+function DashboardScreen () {
+  const auth = useAtomValue(authAtom);
+  const userProfile = useAtomValue(userProfileAtom);
+  const setError = useSetAtom(errorAtom);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [personTotalTime, setPersonTotalTime] = useState<PersonTotalTime[]>();
+  const { personsApi } = useApi();
+  
+/**
+ * Initialize logged in person's time data.
+ * 
+ */
+  const getPersons = async () => {
+    setIsLoading(true);
+    //NOTE: Trainees do not have time bank entries, therefore a use keycloak ID from another employee for testing.
+    const loggedInPerson = await (await personsApi.listPersons({ active: true })).filter(person => person.keycloakId === userProfile?.id)
+
+    if (loggedInPerson.length > 0) {
+      try {
+        if (auth?.tokenRaw){
+          const fetchedPerson : PersonTotalTime[] = await personsApi.listPersonTotalTime({personId: loggedInPerson[0].id});
+          setPersonTotalTime(fetchedPerson);
+        }
+      } catch (error) {
+        setError(`${ "Person fetch has failed." }, ${ error }`);
+      }
+    }
+    else {
+      setError("Your account does not have any time bank entries.");
+    }
+    setIsLoading(false);
+  }
+
+  useEffect(() => {
+    getPersons();
+  }, [auth])
+  
 
   return (
-    <div>
-      <div>This is where we would put our dashboard! IF WE HAD ONE!</div>
-      <button type="button" onClick={ auth?.logout }>Log out</button>
-    </div>
+    <>{(isLoading)
+      ?<div>Loading...</div>
+      :<div>
+        <div>{personTotalTime?.map((person) => {
+          return (`Your balance is ${Math.trunc(person.balance / 60)} h ${(person.balance % 60) * -1} min`)
```

Comment:
I had a look at the original time bank and this balance logic seems a little different, so I assume you are sure its generating the correct balances (or at least the same as the original time bank was)? Perhaps this balance calculation could be put in its own function which returns the balance string as I think this will probably be reused somewhere

Reply:
Compared the time values to the ones in the previous time bank and they seemed identical